core/src/main/scala/org/apache/spark/sql/policy/policyFunctions.scala

```diff
@@ -49,3 +52,32 @@ case class CurrentUser() extends LeafExpression with CodegenFallback {
 
   override def prettyName: String = "current_user"
 }
+
+
+case class  LdapGroupsOfCurrentUser() extends LeafExpression
+    with CodegenFallback {
+  override def foldable: Boolean = true
+  override def nullable: Boolean = false
+
+  override def dataType: DataType = LdapGroupsOfCurrentUser.dataType
+
+  override def eval(input: InternalRow): Any = {
+    val snappySession = SparkSession.getActiveSession.getOrElse(
+      throw new IllegalStateException("SnappySession unavailable")).asInstanceOf[SnappySession]
+    var owner = snappySession.conf.get(Attribute.USERNAME_ATTR, "")
+
+    owner = IdUtil.getUserAuthorizationId(
+      if (owner.isEmpty) Constant.DEFAULT_SCHEMA
+      else snappySession.sessionState.catalog.formatDatabaseName(owner))
+
+    val array = ExternalStoreUtils.getLdapGroupsForUser(owner).
+       map(UTF8String.fromString(_))
+    ArrayData.toArrayData(array)
+  }
+
+  override def prettyName: String = "current_user_ldap_groups"
+}
+
+object LdapGroupsOfCurrentUser {
+  val dataType: DataType = ArrayType(StringType)
+}
```

Comment:
Where is this used?

Reply:
You mean to ask where this function is being used?

Reply:
If so, this is being used by customer in implementing their Row Level security logic.
Where they want to filter out rows based on condition if the user is part of particular ldap group .